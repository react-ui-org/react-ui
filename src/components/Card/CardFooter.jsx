import PropTypes from 'prop-types';
import React from 'react';
import { withGlobalProps } from '../../provider';
import { isChildrenEmpty } from '../_helpers/isChildrenEmpty';
import { transferProps } from '../_helpers/transferProps';
import styles from './Card.scss';

export const CardFooter = ({
  children,
  ...restProps
}) => {
  if (isChildrenEmpty(children)) {
    return null;
  }

  return (
    <div
      {...transferProps(restProps)}
      className={styles.footer}
    >
      {children}
    </div>
  );
};

CardFooter.defaultProps = {
  children: null,
};

CardFooter.propTypes = {
  /**
   * Card actions, usually buttons.
   */
  children: PropTypes.node,
};

export const CardFooterWithGlobalProps = withGlobalProps(CardFooter, 'CardFooter');

export default CardFooterWithGlobalProps;
