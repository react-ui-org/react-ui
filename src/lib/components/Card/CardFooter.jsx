import PropTypes from 'prop-types';
import React from 'react';
import { withProviderContext } from '../../provider';
import styles from './Card.scss';

export const CardFooter = (props) => {
  const {
    children,
  } = props;

  return (
    <div className={styles.footer}>
      {children}
    </div>
  );
};

CardFooter.propTypes = {
  /**
   * Card actions, usually buttons.
   */
  children: PropTypes.node.isRequired,
};

export const CardFooterWithContext = withProviderContext(CardFooter, 'CardFooter');

export default CardFooterWithContext;
