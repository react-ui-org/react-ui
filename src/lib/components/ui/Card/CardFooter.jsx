import PropTypes from 'prop-types';
import React from 'react';
import styles from './Card.scss';

const CardFooter = (props) => {
  const {
    children,
  } = props;

  return (
    <div className={styles.footer}>
      {children}
    </div>
  );
};

CardFooter.propTypes = {
  children: PropTypes.node.isRequired,
};

export default CardFooter;
