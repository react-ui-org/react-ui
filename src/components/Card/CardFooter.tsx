import React from 'react';
import { transferProps } from '../../helpers/transferProps';
import { withGlobalProps } from '../../providers/globalProps';
import { isChildrenEmpty } from '../_helpers/isChildrenEmpty';
import { CardFooterProps } from './CardFooter.types';
import styles from './Card.module.scss';

export const CardFooter: React.FunctionComponent<CardFooterProps> = ({
  children,
  ...restProps
}) => {
  if (isChildrenEmpty(children)) {
    return null;
  }

  return (
    <div
      {...transferProps(restProps)}
      className={styles.footer}
    >
      {children}
    </div>
  );
};

export const CardFooterWithGlobalProps = withGlobalProps(CardFooter, 'CardFooter');

export default CardFooterWithGlobalProps;
