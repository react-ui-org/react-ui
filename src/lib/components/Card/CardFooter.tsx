import React from 'react';
import { withGlobalProps } from '../../provider';
import { isChildrenEmpty } from '../_helpers/isChildrenEmpty';
import { transferProps } from '../_helpers/transferProps';
import styles from './Card.scss';
import { CardFooterProps } from './Card.types';

export const CardFooter: React.FunctionComponent<CardFooterProps> = ({
  children,
  ...restProps
}) => {
  if (isChildrenEmpty(children)) {
    return null;
  }

  return (
    <div
      {...transferProps(restProps)}
      className={styles.footer}
    >
      {children}
    </div>
  );
};
export const CardFooterWithGlobalProps = withGlobalProps(CardFooter, 'CardFooter');

export default CardFooterWithGlobalProps;
