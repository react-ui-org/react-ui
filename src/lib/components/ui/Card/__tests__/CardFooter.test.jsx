import React from 'react';
import { shallow } from 'enzyme';
import { CardFooter } from '../CardFooter';

describe('rendering', () => {
  it('renders correctly', () => {
    const tree = shallow(<CardFooter><p>Card footer content</p></CardFooter>);

    expect(tree).toMatchSnapshot();
  });
});
