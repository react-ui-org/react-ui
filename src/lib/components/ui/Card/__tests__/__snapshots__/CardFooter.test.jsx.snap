// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering renders correctly 1`] = `
<div
  className="footer"
>
  <p>
    Card footer content
  </p>
</div>
`;
